File: codemap/indexer/testdata/java/src/main/java/com/acme/orders/OrderService.java
package com.acme.orders;

import com.acme.store.Repo;
import com.acme.store.PgRepo;
import static com.acme.orders.Util.clean;

public class OrderService {
    private final Repo<Order> repo;

    public OrderService() {
        this.repo = new PgRepo<>();
    }

    public Order place(Order o) {
        validate(o);
        String t = PgRepo.table();
        return repo.save(clean(o));
    }

    private void validate(Order o) {}
}

File: codemap/indexer/testdata/cs/src/Acme.Orders/OrderService.cs
using Acme.Store;
using static Acme.Orders.Util;

namespace Acme.Orders;

public partial class OrderService
{
    private readonly IRepo _repo;

    public OrderService(IRepo repo)
    {
        _repo = repo ?? new PgRepo();
    }

    public int Count => _repo.Count;

    public object Place(Order o)
    {
        Validate(o);
        Audit(o.Id.Shout());
        var table = PgRepo.Table();
        var id = Clean(o.Id);
        return _repo.Save(o);
    }

    private void Validate(Order o)
    {
        if (o == null) throw new System.ArgumentNullException(nameof(o));
    }
}
